### ISSUE TITLE
Wrong test reported as failing while running tests in parallel

### ISSUE THREAD

**Issue by lolsokje** [pestphp/pest#720]
Given this simple test file;

```
<?php

test('this fails', function () {
    $this->assertTrue(false);
});

test('this passes, but is reported as failing', function () {
    $this->assertTrue(true);
});
```

The output of `./vendor/bin/pest --parallel` is 
![image](https://user-images.githubusercontent.com/12721617/226698941-3212a2ed-9f2d-4a10-ac5e-8da01eb1ef44.png)

Notice the "FAILED  Tests\Feature\ExampleTest > this passes, but is reported as failing".

Instead of reporting the actual failed test, the last test in the file is reported instead. The code snippet however does show the correct failing test.

**Comment by nunomaduro** [comment]
Are you using the latest version of Pest and Collision?

**Comment by lukeraymonddowning** [comment]
Will look into this tomorrow 👀

**Comment by lolsokje** [comment]
> Are you using the latest version of Pest and Collision?

Yup, this example is from a fresh Laravel 10 install, with Pest 2.0 and Collision 7.0.

**Comment by nunomaduro** [comment]
Fixed in v2.2.2.
